Readme.md

```diff
@@ -0,0 +1,171 @@
+
+    ![Action select](assets/images/actions-select-workflow.png)
+
+2. Above the list of workflow runs, select Run workflow.
+
+    ![Action dispatch](assets/images/actions-workflow-dispatch.png)
+
+3. Use the Branch dropdown to select the workflow's branch. Click Run workflow.
+
+    ![Action run](assets/images/actions-manually-run-workflow.png)
+
+### Viewing a workflow
+
+After a workflow run has started, you can see a visualization graph of the run's progress and view each step's activity on GitHub.
+
+1. In the left sidebar, click the workflow you want to see.
+
+    ![Workflow](assets/images/learn-github-actions-workflow.png)
+
+2. Under "Workflow runs", click the name of the run you want to see.
+
+    ![Workflow run](assets/images/learn-github-actions-run.png)
+
+3. Under Jobs or in the visualization graph, click the job you want to see.
+
+    ![Job result](assets/images/overview-actions-result-navigate.png)
+
+4. View the results of each step.
+
+    ![Step result](assets/images/overview-actions-result-updated-2.png)
+
+### Updating environment variables (if needed)
+The environment variables are passed to the caller workflows as either inputs or secrets. 
+Whenever you want to add a new environment variable to your repository and pass it as a secret, you should update your workflow as well. 
+
+For this purpose, you need to perform the following steps. 
+#### 1. Update inputs and secrets in a reusable workflow definition
+```
+on:
+  workflow_call:
+    inputs:
+      input-variable:
+        required: true
+        type: string
+    secrets:
+      secret-variable:
+        required: true
+```
+
+#### 2. Reference the input or secret that you defined in the on key in the previous step.
+
+```
+jobs:
+  reusable_workflow_job:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ./.github/workflows/...
+        with:
+          input-variable: ${{ inputs.input-variable }}
+          secret-variable: ${{ secrets.secret-variable }}
+```
+
+#### 3. Include the secret in the caller workflow. 
+- Use the `with` keyword to pass named inputs.
+- Use the `secrets` keyword to pass named secrets. 
+````
+jobs:
+  call-workflow-passing-data:
+    uses: octo-org/example-repo/.github/workflows/reusable-workflow.yml@main
+    with:
+      input-variable: string
+    secrets:
+      secret-variable: ${{ secrets.SECRET_VARIABLE }}
+````
+
+#### 4. Create secrets
+Create a new secret for the specific repository, with the same name you specified in the previous step.
+1. Navigate to the main page of the repository. 
+2. Under the repository name, click Settings
+
+    ![Settings tab](assets/images/repo-actions-settings.png)
+
+3. In the left sidebar, click  Secrets > Actions.
+
+    ![Secrets ](assets/images/secrets.png)
+4. Click New repository secret.
+5. Type a name for your secret in the Name input box.
+
+    ![New secret](assets/images/settings-new-secret.png)
+6. Enter the value for your secret.
+7. Click Add secret.
```

Comment:
Refer to github tutorials?

Reply:
https://docs.github.com/en/enterprise-cloud@latest/actions/security-guides/encrypted-secrets
This is the link to the documentation